### PR TITLE
Fix syntax error when delimited column alias is used without preceding whitespace

### PR DESCRIPTION
**Pull request by Anikait143**
### 1. Issue
When the alias is directly adjacent to a keyword (like END) or a non-delimited column name, the error message mentions AS, even though there is no AS keyword in the query.

### 2. Changes made to fix the issue
- This bug is caused by ANTLR rewriting, we have added an extra space before 'AS' keyword when rewriting the query which which doesn't explicitly has 'AS' keyword before alias. 
```
# Input
SELECT CASE WHEN 1=1  THEN 1 ELSE 0 END[c]
```
```
# previous output (wrong output)               
Msg 33557097, Level 16, State 1, Server persist-rcv-07mar-16-2-0-9, Line 1
syntax error at or near "ENDAS"
```
```
# current output (correct output)
~~START~~
int
1
~~END~~
```

```
# previous output (wrong output)
SELECT 1c;

Msg 33557097, Level 16, State 1, Server persist-rcv-07mar-16-2-0-9, Line 1
trailing junk after numeric literal at or near "1A"
```
```
# current output (correct output)
SELECT 1c;

~~START~~
int
1
~~END~~
```
- Added Test Cases for this issue

Task: BABEL-4863
Authored-by: Anikait Agrawal [agraani@amazon.com](mailto:agraani@amazon.com)
Signed-off-by: Anikait Agrawal [agraani@amazon.com](mailto:agraani@amazon.com)

### Test Scenarios Covered ###
* **Use case based -**   BABEL-4863-vu-*

* **Boundary conditions -**

* **Arbitrary inputs -**

* **Negative test cases -**

* **Minor version upgrade tests -** 

* **Major version upgrade tests -** 

* **Performance tests -**

* **Tooling impact -**

* **Memory tests -**

* **Client tests -**

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/JDBC/input/BABEL-4863-vu-verify.sql [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+select @v[c]
+GO
+declare @v int=1
+select @v'c'
+GO
+declare @v int=1
+select @v"c"
+GO
+select babel_4863_func()[a];
+GO
```
Comment (robverschoor) [inline test/JDBC/input/BABEL-4863-vu-verify.sql:39]: Pls add these test cases as well:

- select 1[c], select 1'c', select 1"c"  (i.e. with a numeric constant)
- select 'abc'[def] and variants (currently this works OK, include a test case to make sure it keeps working)
- include cases where such syntax cases occur in a procedure, function and view
- provide upgrade tests for procs/functions/views, for all releases

File: test/JDBC/input/BABEL-4863-vu-verify.sql [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+SELECT 0.0a;
+GO
+SELECT .0a;
+GO
+SELECT 0.0e1a;
+GO
+SELECT $1a;
+GO
+SELECT CASE WHEN 1=1  THEN 1 ELSE 0 END[c]
```
Comment (robverschoor) [inline test/JDBC/input/BABEL-4863-vu-verify.sql:17]: also include test cases where the delimited identifier includes spaces or special chars, i.e. select 123[this is a test]

**Review by robverschoor — CHANGES_REQUESTED** [review]

File: test/JDBC/input/BABEL-4863-vu-verify.sql (hunk 1 above)
Comment (Anikait143) [reply test/JDBC/input/BABEL-4863-vu-verify.sql:39]: Done, Added Tests and updated upgrade files

File: test/JDBC/input/BABEL-4863-vu-verify.sql (hunk 2 above)
Comment (Anikait143) [reply test/JDBC/input/BABEL-4863-vu-verify.sql:17]: Done, Added Tests

**Review by robverschoor — APPROVED** [review]

**Review by jsudrik — APPROVED** [review]
Cherry pick from the other branch. Approved.

**Pull request merged**